File: tests/pages/reimbursements_page.py
"""
Reimbursements Workspace Page Object (/reimbursements).
"""

from playwright.sync_api import Page, expect
from tests.pages.base_page import BasePage
from tests.components.modal import ModalComponent


class ReimbursementsPage(BasePage):
    PATH = "/reimbursements"

    def __init__(self, page: Page):
        super().__init__(page)
        self.new_claim_btn = (
            page.locator("button")
            .filter(has_text="New Expense Claim")
            .or_(page.locator("button").filter(has_text="Submit Claim"))
            .first
        )
        self.modal = ModalComponent(page)

        # Claim form
        self.title_input = page.locator("input[name='title'], input[placeholder*='Title']").first
        self.amount_input = page.locator("input[name='amount'], input[type='number']").first
        self.category_select = page.locator("select[name='category'], select").first
        self.submit_claim_btn = (
            page.locator("button[type='submit']").or_(page.locator("button").filter(has_text="Submit Claim")).first
        )

    def submit_expense_claim(self, title: str, amount: str, category_idx: int = 1):
        """Creates and submits a new expense reimbursement claim."""
        expect(self.new_claim_btn).to_be_visible()
        self.new_claim_btn.click()
        if self.title_input.is_visible():
            self.title_input.fill(title)
        if self.amount_input.is_visible():
            self.amount_input.fill(amount)
        if self.category_select.is_visible():
            self.category_select.select_option(index=category_idx)
        expect(self.submit_claim_btn).to_be_enabled()
        self.submit_claim_btn.click()

File: e2e/pages/ReimbursementsPage.ts
import { Page, Locator, expect } from "@playwright/test";
import { BasePage } from "./BasePage";

export class ReimbursementsPage extends BasePage {
  readonly categorySelect: Locator;
  readonly dateInput: Locator;
  readonly vendorInput: Locator;
  readonly amountInput: Locator;
  readonly submitButton: Locator;

  constructor(page: Page, baseURL?: string) {
    super(page, baseURL);
    this.categorySelect = page.locator("select").first();
    this.dateInput = page.locator('input[type="date"]').first();
    this.vendorInput = page.locator('input[type="text"]').first();
    this.amountInput = page.locator('input[type="number"]').first();
    this.submitButton = page.locator('button[type="submit"]:has-text("Submit"), button:has-text("Submit Expense")');
  }

  async goto(): Promise<void> {
    await this.navigate("/reimbursements");
  }

  async assertLoaded(): Promise<void> {
    await expect(this.page).toHaveURL(/.*\/reimbursements.*/);
    await expect(this.page.locator("body")).toContainText(/Reimbursement/i);
  }

  async submitClaim(amount: number, vendor: string = "Office Supplies Ltd", dateStr: string = "2026-08-20"): Promise<void> {
    await this.dateInput.fill(dateStr);
    await this.vendorInput.fill(vendor);
    await this.amountInput.fill(String(amount));
    await this.submitButton.click();
  }
}
